client/state/notices/actions.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import impureLodash from 'calypso/lib/impure-lodash';
+
+/**
+ * Internal dependencies
+ */
+import { NOTICE_CREATE, NOTICE_REMOVE } from 'calypso/state/action-types';
+import { extendAction } from 'calypso/state/utils';
+import type {
+	NoticeActionCreator,
+	NoticeActionCreatorWithStatus,
+	NoticeRemovalActionCreator,
+} from 'calypso/state/notices/types';
+
+import 'calypso/state/notices/init';
+
+const { uniqueId } = impureLodash;
+
+export const removeNotice: NoticeRemovalActionCreator = ( noticeId ) => {
+	return {
+		noticeId,
+		type: NOTICE_REMOVE,
+	};
+};
+
+export const createNotice: NoticeActionCreatorWithStatus = (
+	status,
+	text,
+	{ id, ...noticeOptions } = {}
+) => {
+	return {
+		type: NOTICE_CREATE,
+		notice: Object.assign( { showDismiss: true }, noticeOptions, {
+			noticeId: id || uniqueId(),
+			status,
+			text,
+		} ),
+	};
+};
+
+export const successNotice: NoticeActionCreator = ( text, noticeOptions ) =>
+	createNotice( 'is-success', text, noticeOptions );
+export const errorNotice: NoticeActionCreator = ( text, noticeOptions ) =>
+	createNotice( 'is-error', text, noticeOptions );
+export const infoNotice: NoticeActionCreator = ( text, noticeOptions ) =>
+	createNotice( 'is-info', text, noticeOptions );
+export const warningNotice: NoticeActionCreator = ( text, noticeOptions ) =>
+	createNotice( 'is-warning', text, noticeOptions );
+export const plainNotice: NoticeActionCreator = ( text, noticeOptions ) =>
+	createNotice( 'is-plain', text, noticeOptions );
+
+// Higher-order action creator: modify the wrapped creator to return actions with the
+// `notices.skip` meta, so that it's ignored by the notices middleware.
+export const withoutNotice = ( actionCreator ) => ( ...args ) =>
```

Comment:
Is the type missing for this one?

Reply:
Yup 👍  Added in https://github.com/Automattic/wp-calypso/pull/49461/commits/ccb4b4949071fcbe1401d698c2726690b0a1723d

Note that this type likely needs to live somewhere else (in `/state/types.ts` for example) but I'm leaving that for another time. Typing all of `state` and figuring out types that fit well for all of Calypso's state feels like a separate task.

Reply:
Scratch that, I'll need to improve this bit.

Reply:
@saramarcondes I'll have to type `extendAction` too to get that done properly. Given this one is used very little throughout the codebase, I think I'll leave it for another time if you don't mind.

Reply:
Sounds good to me.